### ISSUE TITLE
Meetings not showing

### ISSUE THREAD

**Issue by julien-paris** [code4recovery/12-step-meeting-list#247]
**Describe the bug**

Hi folks,
Following a server migration I ran into some trouble with the meeting guide.
All meetings are still in the DB and accessible through their single URLs; 
however, the meetings page does not list the info.
Please see here: https://www.aaparis.org/meetings/
I have also deactivated/deleted/reinstalled the Meeting guide plugin, to no avail. 
Please advise.
Kind regards, and thanks so much for your amazing service.
Julien 

**To Reproduce**
1. Go to https://www.aaparis.org/meetings/

**Expected behavior**
The meetings should show

**Comment by tech2serve** [comment]
For some reason, the plugin is not completely installed/activated/whatever.

In my browser, when I look at the file structure of the loaded sources, they look like this:

<img width="191" alt="image" src="https://user-images.githubusercontent.com/31904316/102498197-a8931800-4047-11eb-8d5d-f920631df9f4.png">

It should look like this:

<img width="180" alt="image" src="https://user-images.githubusercontent.com/31904316/102498349-daa47a00-4047-11eb-9d51-7252bb54dac2.png">

Your site doesn't have the necessary javascript from the plugin loaded, and no code is being run to list the meetings.

If with your new server you have direct access to the file structure, navigate into the plugins directory and the `12-step-meeting-list` folder. Inside of that, you should have an `assets/js` folder. 

If that folder is missing, then the installation has failed for some reason. You should focus your efforts there first.

If you do have that folder, then something in the loading process is broken. In this case, I'd look into themes or other plugins that could be creating a conflict. Are you using a child theme?

**Comment by julien-paris** [comment]
Thx for the answer. 

The folders are there, so that brings us to your second hypothesis. 

I have tested deactivating all plugins > did not resolve.

We use the 2017 theme, no child theme.

**Comment by tech2serve** [comment]
I'm suspicious then of whatever changes you made display this:

<img width="1215" alt="image" src="https://user-images.githubusercontent.com/31904316/102548175-40b0f180-4088-11eb-90f5-1f7d75d7928a.png">

If you remove it, does the problem persist?

If that doesn't help, and you haven't make any modifications to TSML plugin files, I guess I would next clear the website cache, and perhaps talk with the hosting provider. 

What Wordpress version is loaded? 

What PHP version is your new server loaded with?

Are you experiencing any issues on the admin side? For example, are you able to add a meeting and geocode an address?

**Comment by julien-paris** [comment]
This text is added in the widget 
![image](https://user-images.githubusercontent.com/73740354/102549814-0b72c680-40bd-11eb-8eb9-d8f18e94e3ea.png)

- Tried removing it and it doesn't resolve the issue
- have not made any modifications to the plugin files

This problem has appeared since I migrated, specifically to have latest php/wp versions, which are:

PHP: 7.4.11
WP: 5.6

The logs do show this:
JS: showing 2 of 2
JS: 2020-05-27 20:51:21 [error X 2][http://www.aaparis.org/wp-content/themes/twentyseventeen/assets/js/global.js?ver=20190121:124:33] Cannot read property \'top\' of undefined 
JS: 2020-07-06 11:23:21 [error X 1][http://www.aaparis.org/wp-content/plugins/elementor/assets/js/editor-document.min.js?ver=2.9.13:2:195519] Cannot read property \'map\' of undefined 
PHP: showing 1 of 1
PHP: 2020-07-06 11:11:56 [notice X 3][/htdocs/public/www/wp-content/plugins/elementor/core/common/modules/ajax/module.php::171] Undefined index: data [array (
  'trace' => '
#0: Elementor\Core\Logger\Manager -> shutdown()
',
)]

Yesterday I did test adding a meeting > worked fine. Also, it appeared in the "Upcoming meetings" widget.

I just tried the geocoding, works fine:

![image](https://user-images.githubusercontent.com/73740354/102550485-2eea4100-40be-11eb-8bc7-565b31653aaa.png)

**Comment by tech2serve** [comment]
Ok. I'm going to try and get some others from our team to look at this. 

I also added a text widget on the top of my dev meeting list, and didn't see any issues. I did notice the text widget you have includes "important" as an attribute, and I'm not sure how or why that is different (I'm using the same theme as you). Nonetheless, you've removed that text and same results, so that probably isn't worth a lot of time.

Frankly, the best I can tell you is that something in your new site, be it a plugin, your use of the widget, something with the Wordpress install, or something with the hosting provider's configuration -- frankly, I'm at a loss -- but that page is not loading the javascript.

I'd still suggest clearing the website cache and checking with your hosting provider as we've seen the cache be an issue in the past.

The other thing you can do, which I've have serious doubts it will help, is to navigate to Settings->Permalink, and then simply Save Changes. 

I'm also curious, if you navigate to your Import & Settings and "Where's my Info", then click on "right here"...

<img width="273" alt="image" src="https://user-images.githubusercontent.com/31904316/102553387-c20c8200-4090-11eb-8fdf-182cb04b33b3.png">

...what happens?

**Comment by julien-paris** [comment]
Thx. Yes will try and also investigate with my host but support is terrible so I don't expect to get an answer at all. 
Am also at a loss. Am so tempted to do a full clean wp insall.
Clearing cache: F12 > right click on refresh > Empty cache and Hard Reload : done
I just added W3 to flush cache and I am **now getting an error** message on the meetings page (only). Will investigate.
The permalinks thing I did yesterday and today > nothing there. 
 Import & Settings > "Where's my Info". I had also checked. It sends to the same meetings page. Nothing funny there.

**Comment by julien-paris** [comment]
I turned on debug mode and get this: 
Warning: mysqli_real_connect(): Headers and client library minor version mismatch. Headers:50649 Library:50528 in /home/y645ggme/public_html/wp-includes/wp-db.php on line 1653

**Comment by 97eighty** [comment]
Julien,

Did you reset Permalinks?

**Comment by 97eighty** [comment]
Sorry, looks like you did. Just got through all the message.

**Comment by tech2serve** [comment]
> I turned on debug mode and get this:
> Warning: mysqli_real_connect(): Headers and client library minor version mismatch. Headers:50649 Library:50528 in /home/y645ggme/public_html/wp-includes/wp-db.php on line 1653
> 
> Which seems to indicate a php/mysql version problem. Which would make sense since the prob came up after a major leap in php versions.

Interesting. I'm even more thinking now you need to engage your hosting provider...assuming they're the ones deciding php and mysql versions.

**Comment by julien-paris** [comment]
Indeed I have

**Comment by julien-paris** [comment]
> Julien,
> 
> Did you reset Permalinks?

Indeed I have

**Comment by julien-paris** [comment]
> > I turned on debug mode and get this:
> > Warning: mysqli_real_connect(): Headers and client library minor version mismatch. Headers:50649 Library:50528 in /home/y645ggme/public_html/wp-includes/wp-db.php on line 1653
> > Which seems to indicate a php/mysql version problem. Which would make sense since the prob came up after a major leap in php versions.
> 
> Interesting. I'm even more thinking now you need to engage your hosting provider...assuming they're the ones deciding php and mysql versions.

Will do, although again, I am not getting my hopes up :/

**Comment by 97eighty** [comment]
@julien-paris Send over a sharing key to tsml@code4recovery.org and I'll pull in your meetings. I have a fresh WP install to test.

**Comment by julien-paris** [comment]
> @julien-paris Send over a sharing key to [tsml@code4recovery.org](mailto:tsml@code4recovery.org) and I'll pull in your meetings. I have a fresh WP install to test.

Sorry what is a sharing key?

**Comment by 97eighty** [comment]
@julien-paris Go to Meetings->Import & Settings. Right had side contains a way to create authorized apps. Create a new one and send us the URL it creates. Will look like: https://domain.org/wp-admin/admin-ajax.php?action=meetings&key=908d65a023c25d59de6aa69f53969e9b

**Comment by julien-paris** [comment]
Sorry, I dont see it:

![image](https://user-images.githubusercontent.com/73740354/102560429-84304d80-40d2-11eb-94c3-c8ba53e36ba4.png)

**Comment by julien-paris** [comment]
Ok I see, it appears when the sharing is restricted. 
Our data being shared with the regional site, i will leave it as such if that's ok. 
I have sent you our link to our public data source by email. 

Let me know if that works.

**Comment by 97eighty** [comment]
Doesn't appear that anything is wrong with your meetings -> http://www.97eighty.com.php72-4.phx1-1.websitetestlink.com/meetings/

**Comment by julien-paris** [comment]
Good to know. 
Other indications of that were that they appeared in the widgets properly, and the pdf list document also generates correctly. 
Also, all the meetings are intact when accessed through their single listing page. 
So the only thing not working is the meetings page itself.

**Comment by 97eighty** [comment]
@julien-paris I've identified the PHP function causing the error. In includes/function.php we use the mb_strtolower() function which throws an error in your environment. I've commented out the function on your particular install until I complete more research on cause. For now your meetings are working.

# Unicode-aware lowercase of characters in string
//	return mb_strtolower($t);
	return ($t);

**Comment by 97eighty** [comment]
@julien-paris Your PHP environment was not compiled using the mbstring module. It is disabled. mbstring provides multibyte specific string functions that deal with multibyte encodings in PHP. Some of your meetings have multibyte strings in their meeting names and street names. I suggest asking your hosting provider to enable this module.

**Comment by julien-paris** [comment]
This is great. Thanks so much for taking the time. 
I will send that to my host. (As expected they still haven answered my first ticket).

**Comment by 97eighty** [comment]
The mbstring module is not included by default when building PHP. See https://www.php.net/mbstring. You can use phpinfo() to see if the module is loaded.
